packages/api/src/controllers/access-control.ts

```diff
@@ -8,4 +14,66 @@ const app = Router();
 accessControl.use("/signing-key", signingKeyApp);
 app.use("/access-control", accessControl);
 
+accessControl.post(
+  "/gate",
+  authorizer({ anyAdmin: true }),
+  validatePost("access-control-gate-payload"),
+  async (req, res) => {
+    const playbackId = req.body.stream.replace("video+", "");
+    const content =
+      (await db.stream.getByPlaybackId(playbackId)) ||
+      (await db.asset.getByPlaybackId(playbackId));
+
+    if (!content) {
+      throw new ForbiddenError("Content not found");
+    }
+
+    if (content.deleted) {
+      throw new ForbiddenError("Stream is deleted");
+    }
+
+    const playbackPolicy = content.playbackPolicy;
+
+    if (playbackPolicy) {
+      if (playbackPolicy.type === "public") {
+        res.set("Cache-Control", "max-age=120,stale-while-revalidate=600");
+        res.status(204);
+        return res.end();
+      }
+
+      if (playbackPolicy.type === "signed") {
+        if (!req.body.pub) {
+          throw new ForbiddenError("Stream is gated and requires a public key");
+        }
+
+        const query = [];
+        query.push(sql`signing_key.data->>'publicKey' = ${req.body.pub}`);
+        const [signingKeyOutput] = await db.signingKey.find(query, {
+          limit: 1,
```

Comment:
Can we do `limit: 2` and also make sure there are no signing keys with the same public key? Just a special case below of `output.length > 1` as well. We definitely want to know if we ever have a bug where this happens, and should not allow the repeated signing keys to be used anymore as they are likely compromised.

Reply:
Good point!